src/hooks/useOwners.ts

```diff
@@ -0,0 +1,34 @@
+"use client";
+
+import { SafeAccountV0_3_0 } from "abstractionkit";
+import { useCallback, useEffect, useState } from "react";
+import { useAccount, useClient } from "wagmi";
+import { Address } from "viem";
+
+export function useOwners() {
+  const client = useClient();
+  const account = useAccount();
+  const [owners, setOwners] = useState<Address[] | undefined>(undefined);
+  const [error, setError] = useState<Error | null>(null);
+
+  const listOwners = useCallback(async () => {
+    if (!account?.address || !client?.transport?.url) return undefined;
+
+    try {
+      const safeAccount = new SafeAccountV0_3_0(account.address);
+      const owners = (await safeAccount.getOwners(
+        client.transport.url
+      )) as Address[];
+      return owners;
+    } catch (e) {
+      setError(new Error(`Error listing owners: ${(e as Error).message}`));
+      return undefined;
+    }
+  }, [account?.address, client?.transport?.url]);
+
+  useEffect(() => {
+    listOwners().then((newOwners) => setOwners(newOwners));
+  }, [listOwners]);
+
+  return { owners, error };
+}
```

Comment:
`#should` be consistent with using react-query elsewhere and in turn make your implementation simpler

```suggestion
export function useOwners() {
  const client = useClient();
  const { address } = useAccount();

  const {
    data: owners,
    error,
    isLoading,
  } = useQuery({
    queryKey: ['owners', address],
    queryFn: async () => {
      if (!address || !client?.transport?.url) {
        return undefined;
      }

      const safeAccount = new SafeAccountV0_3_0(address);
      return safeAccount.getOwners(
        client.transport.url
      ) as Promise<Address[]>;
    },
    enabled: Boolean(address && client?.transport?.url),
  });

  return {
    owners,
    error: error ? new Error(`Error listing owners: ${error.message}`) : null,
    isLoading,
  };
}
```